src/search.coffee

```diff
@@ -81,28 +81,9 @@ class Search
       # Filtered searches' data lives within the webPages property.
       body = body.webPages if body._type is 'SearchResponse'
 
-      # Empty responses can cause errors, and should be returned right away.
-      unless body
-        return callback null,
-          estimatedCount: 0
-          results: []
-
-      # Parse an ID out of result URLs for compatibility and duplicate checks.
-      invalidId = false
-      body.value.forEach (result) ->
-        matches = (result.url or result.hostPageUrl).match /&h=([^&]+)/
-        if matches
-          result.id = matches[1]
-        else
-          invalidId = true
-      if invalidId
-        err = _.extend new Error('Unable to parse an ID out of result URL.'),
-          url: result.url or result.hostPageUrl
```

Comment:
Ironically enough, we were throwing an uncaught fatal exception here because -- duh, Sam -- `result` is undefined at this step. Either way, searches would've been returning an error...